base_layer/core/src/base_node/sync/rpc/service.rs

```diff
@@ -137,13 +140,18 @@ impl<B: BlockchainBackend + 'static> BaseNodeSyncService for BaseNodeSyncRpcServ
             )));
         }
 
-        if start_height > metadata.height_of_longest_chain() {
-            return Ok(Streaming::empty());
-        }
         let hash = message
             .end_hash
             .try_into()
             .map_err(|_| RpcStatus::bad_request(&"Malformed end hash received".to_string()))?;
+        if db.fetch_block_by_hash(hash, true).await.is_err() {
+            return Err(RpcStatus::not_found("Requested end block sync hash was not found"));
+        }
+
+        if start_height > metadata.height_of_longest_chain() {
+            return Ok(Streaming::empty());
+        }
```

Comment:
```suggestion
```
We can remove this, it wont ever get to this. 
We added checks for the start and end blocks, so we know we have them in our chain.

Reply:
There is a specific unit test for it, that tests if empty is returned if end < start where both are valid.